src/executor.rs

```diff
@@ -0,0 +1,412 @@
+        if covered != self.last_covered {
+            self.last_covered = covered;
+            // send the coverage stats to the event manager for use in the monitor
+            update_stats(
+                state,
+                event_manager,
+                "wuppiefuzz_code_coverage",
+                UserStatsValue::Ratio(covered, total),
+            );
+        }
+
+        if e_covered != self.last_endpoint_covered {
+            self.last_endpoint_covered = e_covered;
+            // send the coverage stats to the event manager for use in the monitor
+            update_stats(
+                state,
+                event_manager,
+                "wuppiefuzz_endpoint_coverage",
+                UserStatsValue::Ratio(e_covered, e_total),
+            );
+        }
+
+        let current_time = Instant::now();
+        let diff = current_time.duration_since(self.last_window_time).as_secs();
+        if diff > CLIENT_STATS_TIME_WINDOW_SECS {
+            self.last_window_time = current_time;
+            // send the request stats to the event manager for use in the monitor
+            update_stats(
+                state,
+                event_manager,
+                "requests",
+                UserStatsValue::Number(self.performed_requests),
+            );
+        }
+
+        self.reporter
+            .report_coverage(covered, total, e_covered, e_total);
+
+        // If we interrupt using ctrl+c or the timeout is over, request stop!
+        if self.manual_interrupt.load(Ordering::Relaxed)
+            | self
+                .maybe_timeout_secs
+                .map(|timeout| Instant::now() - self.starting_time > timeout)
+                .unwrap_or(false)
+        {
+            if let Err(e) = event_manager.fire(state, Event::Stop) {
+                error!("Err: failed to fire event{:?}", e);
+            }
+            state.request_stop();
+        }
+    }
+
+    /// Uses the embedded coverage clients to generate a coverage report
+    pub fn generate_coverage_report(&self, report_path: &std::path::Path) {
+        self.endpoint_client.generate_coverage_report(report_path);
+        self.coverage_client.generate_coverage_report(report_path);
+    }
+}
+
+impl<EM, FZ, OT> Executor<EM, FZ> for SequenceExecutor<'_, OT>
+where
+    FZ: UsesState<State = FuzzerState>,
+    EM: UsesState<State = FuzzerState>
+        + EventFirer<State = FuzzerState>
+        + EventRestarter
+        + EventProcessor<EM, FZ>,
+    OT: ObserversTuple<OpenApiInput, FuzzerState>,
+{
+    fn run_target(
+        &mut self,
+        _fuzzer: &mut FZ,
+        state: &mut <Self as UsesState>::State,
+        event_manager: &mut EM,
+        input: &<Self as libafl::inputs::UsesInput>::Input,
+    ) -> Result<ExitKind, libafl::Error> {
+        *state.executions_mut() += 1;
+
+        if let Err(Error::ShuttingDown) = self.pre_exec(state, input, event_manager) {
+            return Err(Error::ShuttingDown);
+        }
```

Comment:
Only update executions after actually executing, right?

```suggestion
        if let Err(Error::ShuttingDown) = self.pre_exec(state, input, event_manager) {
            return Err(Error::ShuttingDown);
        }
        
        *state.executions_mut() += 1;
```

Maybe we should move it all the way to the bottom after `post_exec`? Or maybe even in `post_exec`? Thoughts?

Reply:
I put it after the harness call. It's an execution, so post-exec (which happens after the exec(ution)) should see the incremented value :)